tests_php81/ORM/Functional/Ticket/GH10132Test.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests_PHP81\ORM\Functional\Ticket;
+
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\Common\Collections\Collection;
+use Doctrine\Tests\Models\Enums\Suit;
+use Doctrine\Tests\OrmFunctionalTestCase;
+
+class GH10132Test extends OrmFunctionalTestCase
+{
+    public function setUp(): void
+    {
+        parent::setUp();
+
+        $this->createSchemaForModels(
+            Complex::class,
+            ComplexChild::class
+        );
+    }
+
+    public function testQueryBackedEnumInCompositeKeyJoin(): void
+    {
+        $complex = new Complex();
+        $complex->setType(Suit::Clubs);
+
+        $complexChild = new ComplexChild();
+        $complexChild->setComplex($complex);
+
+        $this->_em->persist($complex);
+        $this->_em->persist($complexChild);
+        $this->_em->flush();
+        $this->_em->clear();
+
+        $qb = $this->_em->createQueryBuilder();
+        $qb->select('s')
+            ->from(ComplexChild::class, 's')
+            ->where('s.complexType = :complexType');
+
+        $qb->setParameter('complexType', Suit::Clubs);
+
+        self::assertNotNull($qb->getQuery()->getOneOrNullResult());
+    }
+}
+
+/** @Entity */
```

Comment:
There is indeed no import statement for this… why does the test suite pass when run on Github :thinking:

Reply:
:bulb: I think you need to modify all of these files: https://github.com/doctrine/orm/tree/2.15.x/ci/github/phpunit